### ISSUE TITLE
Proposal to make brace-style fixable

### ISSUE THREAD

**Issue by jrsearles** [eslint/eslint#7074]
**What version of ESLint are you using?**
3.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
N/A

**What did you do? Please include the actual source code causing the issue.**
Run eslint with fixable flag

**What did you expect to happen?**
Expect brace rule violations to be automatically fixed

**What actually happened? Please include the actual, raw output from ESLint.**
N/A

**Comment by jrsearles** [comment]
This is a follow up to #6828 to allow the brace-style violations to be fixable, since these are stylistic in nature.

**Comment by ilyavolodin** [comment]
:+1:

**Comment by platinumazure** [comment]
:+1:, let's do this.

**Comment by mysticatea** [comment]
:+1:

It needs to not remove comments.

```js
/*eslint brace-style: [error, 1tbs]*/

if (a) // comment
{
}
// comment
else {
}
```

**Comment by not-an-aardvark** [comment]
I'll champion this proposal -- I think we should definitely do this.
